enhancements/declarative-index-config.md

```diff
@@ -173,25 +178,26 @@ $ cat etcd.json
     "version": "0.9.2",
     "properties":[
         {
-            "name": "olm.package",
-            "values": {
+            "type": "olm.package.provided",
+            "value": {
                 "packageName": "etcd",
                 "version": "0.9.2"
             }
         },
         {
-            "name":"olm.gvk",
-            "type": "provided",
-            "values":{
-                "group": "etcd.database.coreos.com", 
-                "kind": "EtcdRestore", 
+            "type": "olm.gvk.provided",
```

Comment:
This could be a compatibility problem. Catalog operator assumes that if there are any properties, then provided GVKs will all be expressed as properties (https://github.com/operator-framework/operator-lifecycle-manager/blob/d9ff24161ab4f7d4d323afb662a56f453791329c/pkg/controller/registry/resolver/operators.go#L264), so it won't fall back on the first-class provided APIs field if this one is renamed.

Reply:
I was planning on at least having the GRPC server rename `olm.gvk.provided` back to `olm.gvk` (same for `olm.package.provided` back to `olm.package`). I'm also wondering if we should rename, but also keep the new names. Then we could migrate existing clients and/or build new clients to use the new name, while not breaking old clients.

Reply:
Is it a mistake to set a precedent for modifying properties at serve time? I agree with the second option -- including both the old and new -- but I think it should happen at build/validate time to avoid the coupling between versions of builder/server/catalog-operator. A tool that generates bundle blobs from bundle images would output both for now, and there would be a validation rule to indicate [in]compatibility with catalog-operator versions.

Reply:
Agreed. Re-worded text about this to indicate that `olm.gvk` and `olm.package` properties will be handled statically during build/validation.